state/user_store.go

```diff
@@ -105,25 +109,26 @@ func strongMD5PasswordHash(pass, authKey string) []byte {
 	return bottom.Sum(nil)
 }
 
-// SQLiteUserStore stores user feedbag (buddy list), profile, and
-// authentication credentials information in a SQLite database.
-type SQLiteUserStore struct {
-	db *sql.DB
+// UserStore stores user feedbag (buddy list), profile, and
+// authentication credentials information in a database.
+type UserStore struct {
+	db     *sqlx.DB
+	dbType string
 }
 
-// NewSQLiteUserStore creates a new instance of SQLiteUserStore. If the
+// NewUserStore creates a new instance of UserStore. If the
 // database does not already exist, a new one is created with the required
 // schema.
-func NewSQLiteUserStore(dbFilePath string) (*SQLiteUserStore, error) {
-	db, err := sql.Open("sqlite3", dbFilePath)
+func NewUserStore(dbType, connString string) (*UserStore, error) {
```

Comment:
This `dbType` string config is a bit brittle. Let's instead have two factory methods, one for each DB type:

```go
NewPostgresUserStore(connString string) (*SQLUserStore, error)
NewSQLiteUserStore(dbFilePath string)) (*SQLUserStore, error)
```

This makes the choice of DB less ambiguous, and we don't have to abstract the connection settings to work for two different DBs.